src/Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp

```diff
@@ -0,0 +1,313 @@
+#include <Common/typeid_cast.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+#include <Interpreters/ArithmeticOperationsInAgrFuncOptimize.h>
+#include <IO/WriteHelpers.h>
+#include <Parsers/ASTTablesInSelectQuery.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int UNEXPECTED_AST_STRUCTURE;
+}
+
+static constexpr const char * min = "min";
```

Comment:
`static` means different for functions and variables in this context. You need place these constants into nemeless namespace cause they should be visible only in this cpp.

```
namespace
{
   static constexpr const char * min = "min";
   ...
}
```
You may use struct with some name instead of namespace. But do not place constants that should not be global in global scope.